show/main.py

```diff
@@ -117,22 +117,28 @@ def run_command(command, display_cmd=False):
         click.echo(click.style("Command: ", fg='cyan') + click.style(command, fg='green'))
 
     proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
-    (out, err) = proc.communicate()
 
-    try:
-        click.echo(out)
-    except IOError as e:
-        # In our version of Click (v6.6), click.echo() and click.echo_via_pager() do not properly handle
-        # SIGPIPE, and if a pipe is broken before all output is processed (e.g., pipe output to 'head'),
-        # it will result in a stack trace. This is apparently fixed upstream, but for now, we silently
-        # ignore SIGPIPE here.
-        if e.errno == errno.EPIPE:
-            sys.exit(0)
-        else:
-            raise
+    while True:
+        output = proc.stdout.readline()
+        if output == "" and proc.poll() is not None:
+            break
+        if output:
+            try:
+                click.echo(output.rstrip('\n'))
+            except IOError as e:
+                # In our version of Click (v6.6), click.echo() and click.echo_via_pager() do not properly handle
+                # SIGPIPE, and if a pipe is broken before all output is processed (e.g., pipe output to 'head'),
+                # it will result in a stack trace. This is apparently fixed upstream, but for now, we silently
+                # ignore SIGPIPE here.
+                if e.errno == errno.EPIPE:
+                    print("Got here")
```

Comment:
can we get a more descriptive message here?

Reply:
That was a debug message I missed. It is now removed.